include/asset/asset_trait.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <string>
+
+class AssetTrait {
+ public:
+  virtual ~AssetTrait() = default;
+
+  std::string GetName() const noexcept;
+  std::string GetValue() const noexcept;
+
+  bool operator==(const AssetTrait& trait) const;
+  bool operator!=(const AssetTrait& trait) const;
+  bool operator>(const AssetTrait& trait) const;
+  bool operator>=(const AssetTrait& trait) const;
+  bool operator<(const AssetTrait& trait) const;
+  bool operator<=(const AssetTrait& trait) const;
+
+  operator std::string() const;
+
+  static const AssetTrait NO_TRAIT;
```

Comment:
In general, is better to avoid `ALL_CAPS` always to avoid clashes with macros.

Ref -> https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum5-dont-use-all_caps-for-enumerators